pyanaconda/iutil.py

```diff
@@ -1288,3 +1291,49 @@ def _wrapper(self, *args, **kwargs):
         with self._lock:
             return wrapped(self, *args, **kwargs)
     return _wrapper
+
+def find_bootopt_mitigations(opts):
+    """Check if some mitigations are disabled in boot options.
+
+    :param str opts: space delimited additional boot options
+    :returns: a named tupple reporting which mitigations are currently disabled (if any)
+    :rtype: named tuple
+    """
+    opt_set = set(opts.split(" "))
+    no_pti = "nopti" in opt_set
+    no_ibrs = "noibrs" in opt_set
+    no_ibpb = "noibpb" in opt_set
+    return DisabledMitigations(no_pti=no_pti, no_ibrs=no_ibrs, no_ibpb=no_ibpb)
+
+def set_bootopt_mitigations(opts, no_pti, no_ibrs, no_ibpb):
+    """Make it possible to selectively disable some boot option based mitigations.
+
+    :param str opts: space delimited additional boot options
+    :param bool no_pti: disable Page Table Isolation
+    :param bool no_ibrs: disable Indirect Branch Restricted Speculation
+    :param bool no_ibpb: disable Indirect Branch Prediction Barier
+    """
+
+    # create a set of individual options
+    opt_set = set(opts.split(" "))
+
+    # PTI
+    if no_pti:
+        opt_set.add("nopti")
+    elif "nopti" in opt_set:
+        opt_set.remove("nopti")
+
+    # IBRS
+    if no_ibrs:
+        opt_set.add("noibrs")
+    elif "noibrs" in opt_set:
+        opt_set.remove("noibrs")
+
+    # IBPB
+    if no_ibpb:
+        opt_set.add("noibpb")
+    elif "noibpb" in opt_set:
+        opt_set.remove("noibpb")
+
+    # join back to space delimited string and return
+    return " ".join(opt_set)
```

Comment:
I really don't think this should be part of the `iutil.py`. It is not used anywhere else than performance spoke so why it is here. We shouldn't use `ituil.py` as code dump for everything. It is hard to find something there that way.

Reply:
Moved to ui/helpers.py - that looks like a good place for it.